hs628_tools/regression_check.py

```diff
@@ -0,0 +1,189 @@
+
+        # Density plot of the residuals
+        sns.distplot(residuals)
+        plt.xlabel("Residuals")
+        plt.title('Residual Normality Check')
+        plt.show()
+
+    def check_predictors_independence_plt(self, X, y, colnames):
+        """
+        Checks to see the independence of the predictors by plotting a heatmap of the correlations
+            -----------
+            Param:
+                X: 2-D numpy array of the predictors
+                y: 1-D numpy array of the response variables
+                colnames : Column names of the predictors
+            -----------
+            Return:
+                Doesn't return anything, just shows the plot object built
+        """
+        self.colnames = colnames
+        data_df = pd.DataFrame(X, columns=self.colnames)
+
+        # Heatmap of the correlation matrix
+        sns.heatmap(data_df.corr(), cmap='binary', annot=True)
+        plt.suptitle('Heatmap of Correlations')
+        plt.show()
+
+    def regression_metric(self, X, y):
+        """
+        Calculates a new regression_metric for evaluating the model. R-Squared,F-Statistic,
+        Residual Standard error are used from the summary of the model to calculate this metric.
+        Intuitively The RSE represents the average distance of the observed data from the model
+        thus the lower the RSE the better fit and F-value expresses how much of the model has improved
+        (compared to the mean) given the inaccuracy of the model and R-squared.R2 expresses how much
+        of the total variation in the data can be explained by the mode(the regression line).
+
+        Thus a harmonic between RSE/F-statistic and R2 is computed as a new metric.
+        It ranges from 0 - 1. 0 being the worst and 1 being the highest or the best fit.
+        Disclaimer : This is a original metric thought out from my knowledge of Linear regression metrics
+        I have tried my best to think of all edge cases it might not fail, but is not foolproof.
+            -----------
+            Param:
+                X: 2-D numpy array of the predictors
+                y: 1-D numpy array of the response variables
+            -----------
+            Return:
+                Returns a regression metric that defines the overall fit.
+        """
+        X = sm.add_constant(X)
+        result = sm.OLS(y, X).fit()
+
+        # Metrics required to calculate new reg metric extracted from the result object
+        self.rsquared = result.rsquared
+        self.fstatistic = result.fvalue
+        self.fstat_pval = result.f_pvalue
+        self.resid_sumsq = (result.mse_resid) ** 2
+
+        if self.fstat_pval <= 0.05 or self.rsquared >= 0:
+            self.regmetric = (1 / ((1 / self.rsquared) + ((self.resid_sumsq ** 0.5) / self.fstatistic)))
+        else:
+            self.regmetric = 0
+
+        return self.regmetric
+
+if __name__ == '__main__':
+
+    # Taking the input file as a command line argument
+    fn = sys.argv[1]
```

Comment:
You should first check the number of command line arguments actually provided, and if not enough were then outpost a help message and either use a default value or cleanly exit (rather than generate exception as would happen now).

Reply:
Yah make sense of checking the number of arguments, also I think it is needed to raise a exception in the case there isn't a file with the filename.